dashboard/test/models/user_test.rb

```diff
@@ -13,6 +12,36 @@ class UserTest < ActiveSupport::TestCase
     assert_equal '21+', teacher.age
   end
 
+  # Disable this test if and when we do require teachers to complete school data
+  test 'school info should not be validated' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: nil
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+    assert teacher.valid?, teacher.errors.full_messages
+  end
+
+  test 'identical school info should not be duplicated in the database' do
+    school_attributes = {
+      country: 'US',
+      school_type: SchoolInfo::SCHOOL_TYPE_PUBLIC,
+      state: 'CA'
+    }
+    teacher = create :teacher, school_info_attributes: school_attributes
+    teacher.save!
+
+    teacher2 = create :teacher, school_info_attributes: school_attributes
+    teacher2.save!
+
+    attr = teacher.process_school_info_attributes(school_attributes)
+    school_info = SchoolInfo.where(attr).first
+    assert teacher.school_info == school_info && teacher2.school_info == school_info
+    assert SchoolInfo.where(attr).count == 1
```

Comment:
Again, this could be more succinctly and clearly written as
```
assert_creates(SchoolInfo) do
  2.times do |_i|
    assert_creates(User) do
      create :teacher, school_info_attributes: school_attributes
    end
  end
end
```
Though this won't verify that both teacher's school info's are the same and point to the newly created school info. If you want that, add some additional asserts at the end.